zerver/lib/push_notifications.py

```diff
@@ -744,6 +775,40 @@ def initialize_push_notifications() -> None:
             "See https://zulip.readthedocs.io/en/latest/"
             "production/mobile-push-notifications.html"
         )
+        return
+
+    if uses_notification_bouncer():  # nocoverage
+        # If we're using the notification bouncer, check if we can
+        # actually send push notifications.
+
+        try:
+            realms = send_realms_only_to_push_bouncer()
+        except Exception:
+            # TODO: Catch specific exceptions.
+            for realm in Realm.objects.filter(push_notifications_enabled=True):
+                do_set_realm_property(realm, "push_notifications_enabled", False, acting_user=None)
+            return
+
+        # TODO: For now, we interpret a 200 as meaning it's good for
+        # every realm, without actually inspecting any data.
+        for realm_uuid, data in realms:
+            realm = Realm.objects.get(uuid=realm_uuid)
+            do_set_realm_property(
+                realm, "push_notifications_enabled", data["can_push"], acting_user=None
+            )
+            Realm.objects.filter(uuid=realm_uuid).update(
+                push_notifications_enabled_end_timestamp=data["expected_end_timestamp"]
+            )
```

Comment:
We probably want to use `do_set_realm_property` to change this value as well, since we'll want to send events.